File: exmple_js_project/src/config/apiConfig.ts
import { SALES_PORTAL_API_URL } from "./env";

export const apiConfig = {
  baseURL: SALES_PORTAL_API_URL,
  endpoints: {
    products: "/api/products",
    productById: (id: string) => `/api/products/${id}/`,
    productsAll: "/api/products/all",
    login: "/api/login",
    metrics: "/api/metrics",
    customers: "/api/customers",
    customerById: (id: string) => `/api/customers/${id}/`,
    customersAll: "/api/customers/all",
    orders: "/api/orders",
    orderById: (id: string) => `/api/orders/${id}/`,
    ordersAll: "/api/orders/all",
    orderDelivery: (id: string) => `/api/orders/${id}/delivery`,
    orderReceive: (id: string) => `/api/orders/${id}/receive`,
    orderStatus: (id: string) => `/api/orders/${id}/status`,
    orderAssignManager: (orderId?: string, managerId?: string) => `/api/orders/${orderId}/assign-manager/${managerId}`, // data > types > user.types.ts
    orderUnassignManager: (orderId?: string) => `/api/orders/${orderId}/unassign-manager`, // data > types > user.types.ts
    orderComments: (orderId: string) => `/api/orders/${orderId}/comments`,
    orderCommentById: (orderId: string, commentId: string) => `/api/orders/${orderId}/comments/${commentId}`,
    notifications: "/api/notifications",
    notificationAsRead: (id: string) => `/api/notifications/${id}/read`,
    notificationsAllAsRead: "/api/notifications/mark-all-read",
    users: "/api/users",
    userById: (id: string) => `/api/users/${id}/`,
  },
};

File: sales-portal-robot-tests/variables/api_config.py
"""RF Variable file — API endpoint URL constants and builder functions."""
from __future__ import annotations

import os

from dotenv import load_dotenv

load_dotenv(".env.dev" if os.getenv("TEST_ENV") == "dev" else ".env")

_BASE_URL: str = os.environ.get("SALES_PORTAL_API_URL", "http://localhost:8686")

# Static endpoint constants (become RF variables when loaded as variable file)
LOGIN: str = f"{_BASE_URL}/api/login"
LOGOUT: str = f"{_BASE_URL}/api/logout"
PRODUCTS: str = f"{_BASE_URL}/api/products"
PRODUCTS_ALL: str = f"{_BASE_URL}/api/products/all"
CUSTOMERS: str = f"{_BASE_URL}/api/customers"
CUSTOMERS_ALL: str = f"{_BASE_URL}/api/customers/all"
ORDERS: str = f"{_BASE_URL}/api/orders"
NOTIFICATIONS: str = f"{_BASE_URL}/api/notifications"
NOTIFICATIONS_MARK_ALL_READ: str = f"{_BASE_URL}/api/notifications/mark-all-read"
METRICS: str = f"{_BASE_URL}/api/metrics"
USERS: str = f"{_BASE_URL}/api/users"


# Parameterised endpoint builder functions (called from keyword libraries)
def product_by_id(product_id: str) -> str:
    return f"{PRODUCTS}/{product_id}"


def customer_by_id(customer_id: str) -> str:
    return f"{CUSTOMERS}/{customer_id}"


def customer_orders(customer_id: str) -> str:
    return f"{CUSTOMERS}/{customer_id}/orders"


def order_by_id(order_id: str) -> str:
    return f"{ORDERS}/{order_id}"


def order_delivery(order_id: str) -> str:
    return f"{ORDERS}/{order_id}/delivery"


def order_status(order_id: str) -> str:
    return f"{ORDERS}/{order_id}/status"


def order_receive(order_id: str) -> str:
    return f"{ORDERS}/{order_id}/receive"


def order_comments(order_id: str) -> str:
    return f"{ORDERS}/{order_id}/comments"


def order_comment_by_id(order_id: str, comment_id: str) -> str:
    return f"{ORDERS}/{order_id}/comments/{comment_id}"


def assign_manager(order_id: str, manager_id: str) -> str:
    return f"{ORDERS}/{order_id}/assign-manager/{manager_id}"


def unassign_manager(order_id: str) -> str:
    return f"{ORDERS}/{order_id}/unassign-manager"


def notification_read(notification_id: str) -> str:
    return f"{NOTIFICATIONS}/{notification_id}/read"


def user_by_id(user_id: str) -> str:
    return f"{USERS}/{user_id}"


def user_password(user_id: str) -> str:
    return f"{USERS}/password/{user_id}"
